presto-native-execution/presto_cpp/main/connectors/arrow_flight/tests/utils/FlightConnectorTestBase.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include "arrow/flight/api.h"
+#include "presto_cpp/main/connectors/arrow_flight/tests/utils/StaticFlightServer.h"
+#include "velox/common/config/Config.h"
+#include "velox/connectors/Connector.h"
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+
+namespace facebook::presto::connector::afc::test {
+
+static const std::string kFlightConnectorId = "test-flight";
+
+class FlightConnectorTestBase : public velox::exec::test::OperatorTestBase {
+ public:
+  void SetUp() override;
+
+  void TearDown() override;
+
+ protected:
+  explicit FlightConnectorTestBase(
+      std::shared_ptr<velox::config::ConfigBase> config)
+      : config_{config} {}
+
+  FlightConnectorTestBase()
+      : config_{std::make_shared<velox::config::ConfigBase>(
+            std::move(std::unordered_map<std::string, std::string>{}))} {}
+
+ private:
+  std::shared_ptr<velox::config::ConfigBase> config_;
+};
+
+/// Creates and registers an arrow flight connector and
+/// spawns a Flight server for testing
+class FlightWithServerTestBase : public FlightConnectorTestBase {
```

Comment:
In IBM code we have an optional test which validates the connector against an external IBM Flight Server. In this test, we can't extend `FlightWithServerTestBase` because we don't want to spin up a `StaticFlightServer`. We still need to set up the connector, so we extend `FlightConnectorTestBase` directly.

In general, this separation is intended to make it easier to plug out the `StaticFlightServer` with something else.